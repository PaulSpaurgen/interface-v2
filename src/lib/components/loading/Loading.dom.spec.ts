import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/svelte';
import LoaderAnimatedPing from './LoadingAnimatedPing.svelte'; // Replace with the correct path to your component
import LoaderAnimatedModal from './LoadingAnimationModal.svelte'; // Replace with the correct path to your component
import html from '@playpilot/svelte-htm';

describe('Loader Animated Ping', () => {
    it('renders loader with two child div elements', () => {
        const { container } = render(LoaderAnimatedPing);

        const loader = container.querySelector('.marlin-loader');
        expect(loader).toBeTruthy();
        expect(loader?.className).contain('marlin-loader');

        const rippleDivs = container.querySelectorAll('.lds-ripple div');
        // There should be two divs for the ripple effect.
        expect(rippleDivs.length).toBe(2);

        // Snapshot testing to ensure consistent render
        expect(container).toMatchSnapshot();
    });
});

describe('Loader Animated Ping', () => {
    it('displays loading indicator with padding', () => {
        const { container } = render(LoaderAnimatedModal, {
            props: {
                loading: true,
                hasPadding: true
            },
        });

        const loadingElement = container.querySelector('span.relative');
        expect(loadingElement).toBeTruthy();
        expect(loadingElement?.className).contain('h-6 w-6');

        // Snapshot test
        expect(container).toMatchSnapshot();
    });

    it('displays loading indicator without padding', () => {
        const { container } = render(LoaderAnimatedModal, {
            props: {
                loading: true,
                hasPadding: false
            },
        });

        const loadingElement = container.querySelector('span.relative');
        expect(loadingElement).toBeTruthy();
        expect(loadingElement?.className).contain('h-5 w-5');

        // Snapshot test
        expect(container).toMatchSnapshot();
    });

    it('displays content block with padding when not loading', () => {
        const { container } = render(LoaderAnimatedModal, {
            props: {
                loading: false,
                hasPadding: true
            },
        });

        const contentElement = container.querySelector('div.flex');
        expect(contentElement).toBeTruthy();
        expect(contentElement?.className).contain('h-6 w-6');

        // Snapshot test
        expect(container).toMatchSnapshot();
    });

    it('displays content block without padding when not loading', () => {
        const { container } = render(LoaderAnimatedModal, {
            props: {
                loading: false,
                hasPadding: false
            },
        });

        const contentElement = container.querySelector('div.flex');
        expect(contentElement).toBeTruthy();
        expect(contentElement?.className).contain('h-5 w-5');

        // Snapshot test
        expect(container).toMatchSnapshot();
    });



    it("should render component inside slot properly", async () => {
        const { findByText } = render(html`
                <${LoaderAnimatedModal}>
                 <div>content</div>
                </${LoaderAnimatedModal}>
              `);

        expect(await findByText('content')).toBeTruthy();
    });
});
