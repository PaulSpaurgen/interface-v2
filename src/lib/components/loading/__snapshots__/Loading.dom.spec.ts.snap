// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Animated Ping > displays content block with padding when not loading 1`] = `
<body>
  <div>
    <div
      class="flex justify-center rounded-3xl bg-grey-400 text-white h-6 w-6"
    />
    
    <!--&lt;LoadingAnimationModal&gt;-->
  </div>
</body>
`;

exports[`Loader Animated Ping > displays content block without padding when not loading 1`] = `
<body>
  <div>
    <div
      class="flex justify-center rounded-3xl bg-grey-400 text-white h-5 w-5"
    />
    
    <!--&lt;LoadingAnimationModal&gt;-->
  </div>
</body>
`;

exports[`Loader Animated Ping > displays loading indicator with padding 1`] = `
<body>
  <div>
    <span
      class="relative flex justify-center rounded-3xl text-white h-6 w-6"
    >
      <span
        class="absolute inline-flex h-full w-full animate-ping rounded-full bg-orange-400 opacity-75"
      />
       
      <span
        class="relative inline-flex justify-center rounded-full bg-orange-400 text-center text-sm h-6 w-6 p-[2.5px]"
      />
    </span>
    
    <!--&lt;LoadingAnimationModal&gt;-->
  </div>
</body>
`;

exports[`Loader Animated Ping > displays loading indicator without padding 1`] = `
<body>
  <div>
    <span
      class="relative flex justify-center rounded-3xl text-white h-5 w-5"
    >
      <span
        class="absolute inline-flex h-full w-full animate-ping rounded-full bg-orange-400 opacity-75"
      />
       
      <span
        class="relative inline-flex justify-center rounded-full bg-orange-400 text-center text-sm h-5 w-5"
      />
    </span>
    
    <!--&lt;LoadingAnimationModal&gt;-->
  </div>
</body>
`;

exports[`Loader Animated Ping > renders loader with two child div elements 1`] = `
<body>
  <div>
    <div
      class="marlin-loader s-pih5pjlcIrgJ"
    >
      <div
        class="lds-ripple s-pih5pjlcIrgJ"
      >
        <div
          class="s-pih5pjlcIrgJ"
        />
         
        <div
          class="s-pih5pjlcIrgJ"
        />
      </div>
    </div>
    <!--&lt;LoadingAnimatedPing&gt;-->
  </div>
</body>
`;
